File: mooc-2024/part-3/2-working-with-strings/09-a-framed-word/main.py
hor = '******************************'

word = input('Word: ')
padL = (30 - 2 - len(word)) // 2
padR = 30 - 2 - len(word) - padL
print(hor)
print('*' + padL * ' ' + word + padR * ' ' + '*')
print(hor)

File: mooc-2024/part-3/2-working-with-strings/09-a-framed-word/main.c
#include <stdio.h>
#include <string.h>

int main() {
    char hor[] = "******************************";
    char word[28];

    printf("Word: ");
    scanf("%s", word);

    int padL = (30 - 2 - strlen(word)) / 2;
    int padR = 30 - 2 - strlen(word) - padL;

    printf("%s\n", hor);
    printf("*");
    while (padL > 0) {
        padL--;
        printf(" ");
    }
    printf("%s", word);
    while (padR > 0) {
        padR--;
        printf(" ");
    }
    printf("*\n");

    printf("%s\n", hor);
    
    return 0;
}